### DISCUSSION TITLE
My Pockethost instance is idle and not responding

### DISCUSSION THREAD

**Discussion by renatocaliari** [pockethost/pockethost#222] [Q&A] [answered]
My instance stopped working: any endpoint of the API or URL on my admin that I try to access on Pockethost keeps loading indefinitely and doesn't return a result. My instance is marked as "idle".

What could be the issue? Is there anything I can do?

Thanks in advance.

![Screenshot 2023-06-12 10 06 19](https://github.com/benallfree/pockethost/assets/1719651/06dbd166-b44e-4cd5-98ba-53f79f158cb8)
![Screenshot 2023-06-12 10 06 02](https://github.com/benallfree/pockethost/assets/1719651/ab9579af-60df-4926-ae41-8482b1663cca)

**Comment by renatocaliari** [comment]
After more than 20 minutes, my instance started working again, without me doing anything. 
If anyone knows the reason why it was down for so long, please let me know.

**Comment by benallfree** [comment]
@renatocaliari I can confirm that https://chubby-zoo.pockethost.io/_/#/login loads for me.

Please let me know if you experience this again. 

I do see an odd drop in CPU:

<img width="1002" alt="image" src="https://github.com/benallfree/pockethost/assets/1068356/4f1d5cf4-1fd0-4984-8818-8105a1643230">

Does this correspond to the time period you were experiencing issues? Normally CPU is hovering around 15%, so seeing it drop to almost 0% would seem to indicate that nothing was reaching PocketHost.

**Reply by renatocaliari** [reply]
Thank you for the investigation, @benallfree . 👏

Although I don't know where the timezone of this chart is from, I believe it was during that period since it seems to be the total amount of time it stopped working for me.

My instance is working again. :)

Is there any indication of what may have happened?
Add, is there something I can do to prevent or recover from it if it happens again?

**Reply by benallfree** [reply, marked as answer]
I could look through the raw logs, but we are aware of a few "mystery" behaviors where the a nodejs process will become unresponsive. We have been solving this by cycling the node processes periodically, which could explain why the problem went away after 15 or so minutes.

I probably won't investigate this one any further since I know there are some known issues in this area. But if this continues happening after those issues are fixed, we will definitely take a second look.

**Reply by renatocaliari** [reply]
This happened to me again right now (6:27 am, Brazil time). It has been out for a few minutes and it hasn't come back yet. 

And yesterday afternoon I think it happened again as well.

**Comment by ghost** [comment]
Same is happened to me, it started working again after a few minutes.

**Reply by benallfree** [reply]
Thanks for the report.

**Comment by benallfree** [comment]
Guys lets move to #223 please.
